openvino_xai/methods/black_box/rise.py

```diff
@@ -103,7 +103,10 @@ def _run_synchronous_explanation(
         prob: float,
         seed: int,
     ) -> np.ndarray:
-        _, _, height, width = data_preprocessed.shape
+        if data_preprocessed.shape[-1] == 3:  # bhwc layout
```

Comment:
Hope this would be enough criteria ;p

Reply:
Please consider:
```
_, dim0, dim1, dim2 = image.shape
if dim0 > dim2 and dim1 > dim2:  # bhwc layout
```

I would propose to unify this check in a separate function, similar check is used in here
https://github.com/openvinotoolkit/openvino_xai/blob/e2cd66f22ae6a3c3e770015f63aec936103aa94b/openvino_xai/explainer/utils.py#L167